crates/top/rerun/src/commands/entrypoint.rs

```diff
@@ -1309,3 +1246,133 @@ fn stream_to_rrd_on_disk(
 
     Ok(())
 }
+
+/// Describes how to handle URLs passed on the CLI.
+struct UrlParamProcessingConfig {
+    data_sources_from_http_urls: bool,
+    data_sources_from_redap_datasets: bool,
+    data_source_from_filepaths: bool,
+}
+
+impl UrlParamProcessingConfig {
+    /// Instruct to create data sources for everything we can.
+    ///
+    /// This is used for pure servers and file redirects.
+    fn convert_everything_to_data_sources() -> Self {
+        // Write to file makes everything it can a data source.
+        Self {
+            data_sources_from_http_urls: true,
+            data_sources_from_redap_datasets: true,
+            data_source_from_filepaths: true,
+        }
+    }
+
+    #[allow(dead_code)] // May be unused depending on feature flags.
+    fn grpc_server_and_web_viewer() -> Self {
+        // GRPC with web viewer can handle everything except files directly.
+        Self {
+            data_sources_from_http_urls: false,
+            data_sources_from_redap_datasets: false,
+            data_source_from_filepaths: true,
+        }
+    }
```

Comment:
This guy right here is the weird one. If we didn't support loading local file path from web viewer, we would be in a simpler world where we either eagerly (proxy) or lazily (native) convert everything to data sources, which could simplify stuff a bit more.

Do we really have reasons to support this use case?

(caveat: i may not understand correctly any of this and be completely wrong)